Fix downloading extensions

BelleNottelling opened the pull request:
This PR fixes an issue introduced by #861 that prevented the system from downloading extensions from the extension store.
Looking at the docs for the symfony HTTP package, the `buffer` option is a boolean which is used to enable or disable writing to the PHP temp path, and not a way to specify a location to store the contents of the response

admdly commented:
The `buffer` option in http-client accepts a boolean, resource, or Clojure (https://github.com/symfony/symfony/blob/6.2/src/Symfony/Contracts/HttpClient/HttpClientInterface.php#method_withOptions). 

Without a buffer, there could be issues depending on file size as the response content is being held in memory.

BelleNottelling commented:
Updated PR includes the suggested changes by @admdly and also cleans up the extraction / installation steps  so that it's easier to read / maintain. I removed some of the tests because the remaining ones already tested the potential exceptions that could be thrown
